tests/test_utils.py

```diff
@@ -117,6 +120,64 @@ def xy_to_pir(x, y, _image_id):
     )
 
 
+class TestXyToPirApiSingle:
+    @pytest.mark.internet
+    @pytest.mark.parametrize("image_id", EXISTING_IMAGE_IDS)
+    def test_fetching_online_no_mocking(self, image_id):
+        x, y = np.random.randint(0, 10000, 2)
+        p, i, r = xy_to_pir_API_single(x, y, image_id=image_id)
+        assert np.isfinite(p)
+        assert np.isfinite(i)
+        assert np.isfinite(r)
```

Comment:
Why are we not checking the exact values if we do a real requests anyway ?

Reply:
Ah but `x` and `y` are always random, so to get the real PIR value one would be forced to kind of re-implement the logic of the actual function, which I didn't really like to do...

By the way, this `np.isfinite` check was there before but in a slightly different setting.

I think an alternative could be to hard-code `x` and `y` for each of the dataset IDs, but I don't know. What do you think?

Reply:
I see your point! I am just wondering if this test could really detect some bugs by simply checking if the number are finite 🤔

Reply:
I think you're right, these tests are probably not telling us much.

What about this kind of test?
```python
    @pytest.mark.internet
    @pytest.mark.parametrize("image_id", EXISTING_IMAGE_IDS)
    def test_fetching_online_no_mocking(self, image_id):
        x, y = np.random.randint(0, 10000, 2)
        pir = xy_to_pir_API_single(x, y, image_id=image_id)
        assert len(pir) == 3
        assert (isinstance(coordinate, float) for coordinate in pir)
```
But it actually does pretty much the same thing as before - the previous test would also fail if the length isn't 3 or if the coordinates are not numbers.

We can also just skip the asserts and just write
```python
xy_to_pir_API_single(x, y, image_id=image_id)
```
which would just test if the online queries work in principle and don't raise any errors. The returned values and types are then covered by other (offline) tests.

What do you think?

Reply:
You are right! 

I agree, It is already good to test that the URL is still valid and does not raise any error + that we have 3 numbers as output! We can keep as is! Thanks for the reply @Stannislav!